targets/targets.json

```diff
@@ -5024,6 +5024,76 @@
             "DISCO_H747I"
         ]
     },
+    "STM32H745xI": {
```

Comment:
all 3 are base targets? We used to call these `MCU_` to recognize what is actual MCU boilerplate

Reply:
No "official" Mbed-supported boards can be built with this. I have an open-source (unreleased) personal project I'm working on that runs Mbed on the H745 chip on a custom board.

I do not plan to introduce it as an official target since it's for a side project and not a partner-supported target.

That being said, I wanted to contribute my work on supporting the H745 chip on Mbed so others may be able to build their own targets (or even add support for ST H745 dev kits).

I can rename these targets to include the `MCU_` prefix so it's more apparent they are base targets rather than buildable ones.

Reply:
@AGlass0fMilk @0xc0170 @MarceloSalazar 
Could you check https://github.com/jeromecoutant/mbed/tree/STM32H745
and comment!
Thx

Reply:
@LMESTM you can also comment :-)
Thx

Reply:
@jeromecoutant thanks for the proposal. I like the idea of `MCU_STM32H745I` as this will enable others to inherit (including official dev boards).
But I don't think we should introduce `GENERIC_H745I*`. I'd suggest to either remove it or rename it to a dev board (NUCLEO?) that you can use to run regular tests.

Reply:
@MarceloSalazar this would be worth a discussion. We don't have a NUCLEO or DISCO for every MCU type so I like the idea of "GENERIC" targets. Idea is to let anyone with a custom board to run the minimum set of tests by just overloading STDIO pins and a LED default pin, and then derive their own target. If you stick to MCU level, we won't provide have a ready-to-use basic target for H745  ... . I agree that a GENERIC target wouldn't be tested as much as regular boards or not tested sometimes before there is a customer target, but I think that would be added value for our customers....

Reply:
> I agree that a GENERIC target wouldn't be tested as much as regular boards

Generic targets allows compilation at least

Reply:
Can you guys get the ST mothership to send you a STM32H745 Nucleo board to test with? That way we can have a buildable target.

Reply:
> > I agree that a GENERIC target wouldn't be tested as much as regular boards
> 
> Generic targets allows compilation at least

Does the CI run tests that just check if a certain target compiles? Even if the target doesn't have a physical board to run tests on?

Reply:
> Can you guys get the ST mothership to send you a STM32H745 Nucleo/Discovery board to test with? That way we can have a buildable target.

Sorry we're taking some time here, but the discussion that it has triggered is wider than H745 MCU alone ... we don't have a NUCLEO or DISCO board for every single MCU and also we can't afford to support all available NUCLEO and DISCO boards, but what really matters is that customers can enable their own custom boards as easily as possible.

Reply:
We're ok to proceed with introducing a Generic target
@jeromecoutant please help us to finalize the review and let's get it done :)